docs/project/contribution_tools.md

```diff
@@ -13,201 +13,241 @@ contributions.
+            periodically run `sudo apt update && sudo apt upgrade`.
+    -   [Homebrew](https://brew.sh/) (for macOS)
+        -   To upgrade versions of `brew` packages, it will be necessary to
+            periodically run `brew upgrade`.
+    -   [`python3` and `pip3`](https://python.org)
+        -   Carbon requires Python 3.9 or newer.
+        -   To upgrade versions of `pip3` packages, it will be necessary to
+            periodically run `pip3 list --outdated`, then
+            `pip3 install -U <package>` to upgrade desired packages.
+        -   When upgrading, version dependencies may mean packages _should_ be
+            outdated, and not be upgraded.
+-   Main tools
+    -   [Bazel](https://www.bazel.build/)
+    -   [Bazelisk](https://docs.bazel.build/versions/master/install-bazelisk.html)
+        (for macOS): Handles Bazel versions.
+    -   [Clang](https://clang.llvm.org/) and [LLVM](https://llvm.org/)
+        -   NOTE: Most LLVM 14+ installs should build Carbon. If you're having
+            issues, see
+            [troubleshooting build issues](#troubleshooting-build-issues).
+    -   [gh CLI](https://github.com/cli/cli): Helps with GitHub.
+    -   [pre-commit](https://pre-commit.com): Validates and cleans up git
+        commits.
+-   Libraries
+    -   zlib1g-dev: Used as a library, but not installed on all Linux systems.
+
+#### Running pre-commit
+
+[pre-commit](https://pre-commit.com) is typically set up using
+`pre-commit install`. When set up in this mode, it will check for issues when
+`git commit` is run. A typical commit workflow looks like:
+
+1. `git commit` to try committing files. This automatically executes
+   `pre-commit run`, which may fail and leave files modified for cleanup.
+2. `git add .` to add the automatically modifications done by `pre-commit`.
+3. `git commit` again.
+
+You can also use `pre-commit run` to check pending changes without `git commit`,
+or `pre-commit run -a` to run on all files in the repository.
+
+> NOTE: Some developers prefer to run `pre-commit` on `git push` instead of
+> `git commit` because they want to commit files as originally authored instead
+> of with pre-commit modifications. To switch, run
+> `pre-commit uninstall && pre-commit install -t pre-push`.
+
+### Optional tools
+
+These tools aren't necessary to contribute to Carbon, but can be worth
+considering if they fit your workflow.
+
+-   [GitHub Desktop](https://desktop.github.com/): A UI for managing GitHub
+    repositories.
+-   `rs-git-fsmonitor` and Watchman: Helps make `git` run faster on large
+    repositories.
+    -   **WARNING**: Bugs in `rs-git-fsmonitor` and/or Watchman can result in
+        `pre-commit` deleting files. If you see files being deleted, disable
+        `rs-git-fsmonitor` with `git config --unset core.fsmonitor`.
+-   [vim-prettier](https://github.com/prettier/vim-prettier): A vim integration
+    for Prettier, which we use for formatting.
```

Comment:
If we're not going to list Prettier itself above, should we do so here? I assume the `vim` plugin depends on it.